content/blog/2024/zoomies.md

```diff
@@ -0,0 +1,464 @@
+
+In most situations,
+the two viewports
+are the same size --
+the size of your browser window,
+or what's left of it after drawing
+the tabs and toolbars.
+But (as we'll see)
+there are some zoomed-in situations
+where the visual viewport
+can end up smaller (but never larger)
+than the layout viewport.
+
+It gets confusing
+(to me at least)
+because both viewports can overflow
+in different ways.
+But the size of our content
+only impacts one of them.
+When we add more content,
+we can overflow the layout viewport.
+In order to overflow
+the visual viewport,
+we'll need browser help!
+
+In researching this article,
+I also came across
+an old two-part QuirksMode post by PPK --
+[_A tale of two viewports_](https://www.quirksmode.org/mobile/viewports.html) --
+which covers this in depth.
+
+## Getting the zoomies
+
+To _zoom_ or _scale_ a page,
+we have to manipulate either the
+size of a CSS pixel
+in relation to the layout viewport,
+or the relationships _between_ the two viewports.
+Each approach gives a different result.
+
+This behavior is defined in
+the CSS [View Module specification](https://drafts.csswg.org/cssom-view-1/#zooming).
+
+### Page zoom: CSS pixels vs. the layout viewport
+
+Browsers all provide
+a _page zoom_ feature for us
+as we surf the web.
+I use it all the time.
+By default,
+pages load at `100%` page zoom,
+but we can zoom in or out from there.
+Generally,
+browsers will remember our zoom settings
+for each domain we visit.
+(I used to have Wikipedia load at 150%,
+but now they provide built-in tools
+for scaling the font size.
+Thanks, Wikipedia!)
+
+This is the most common form
+of zoom available to us web surfers.
+I use the <kbd>Ctrl+/-</kbd>
+(<kbd>Cmd+/-</kbd> on mac)
+keyboard shortcuts quite often,
+but these controls are also available
+in a browser menu.
+
+Page zoom is similar
+to the resolution setting in your operating system.
+Adjusting the page zoom
+will change the ratio of
+_CSS pixels_ vs _device pixels_.
+In fact,
+browsers combine the operating system and page zoom
+to provide an overall _device pixel ratio_ --
+the relationship between a (physical) pixel
+and a rendered (CSS) pixel.
+
+Pixels on my 4k monitor are already zoomed `2x`/`200%`
+by the operating system,
+before the browser gets involved.
+If I also zoom a web page by `2x`/`200%` in the browser,
+the result is a `4x`/`400%` overall zoom --
+and a _device pixel ratio_ of `4:1`.
+
+This _zoom_ is applied to the size of a CSS pixel,
+_before the page is rendered_.
+By zooming in,
+we make each 'pixel' larger.
+But our layout viewport isn't growing at all,
+so our layout now contains _fewer_ `px`
+in each dimension.
+In effect, we've made
+the layout viewport _smaller_
+in relation to our pixels.
+
+**Page Zoom** is adjusting
+_the size of a CSS pixel in relation to the layout viewport_.
+Since that happens before rendering,
+it impacts the layout of the page.
+It's then reflected by media queries,
+which query a 'smaller' viewport.
```

Comment:
This would only be smaller if they zoom in, right? And media queries are one way of observing the layout impact, but it's not the only way, I think? Are there others we could list?